tests/testthat/test-mergingandstackingutilities.R

```diff
@@ -0,0 +1,77 @@
+test_that("variableType", {
+    expect_equal(variableType(variable = 1:10), "Numeric")
+    expect_equal(variableType(variable = LETTERS), "Text")
+    categorical.var <- structure(c(1, 2, 1),
+                                 labels = structure(1:2, .Names = c("A", "B")))
+    expect_equal(variableType(variable = categorical.var), "Categorical")
+    dates <- structure(c(18802, 18803, 18804), class = "Date")
+    expect_equal(variableType(variable = dates), "Date")
+    date.times <- structure(c(1624538040, 1624589160, 1624733100),
+                            class = c("POSIXct", "POSIXt"), tzone = "UTC")
+    expect_equal(variableType(variable = date.times), "Date/Time")
+    duration <- structure(10:12, class = "difftime", units = "mins")
+    expect_equal(variableType(variable = duration), "Duration")
+})
+
+test_that("isDateType", {
+    expect_equal(isDateType(var.types = c("Date", "Date/Time", "Numeric")),
+                 c(TRUE, TRUE, FALSE))
+})
+
+test_that("allIdentical", {
+    expect_equal(allIdentical(x = 1:3), FALSE)
+    expect_equal(allIdentical(x = c(1, 1, 1)), TRUE)
+})
+
+test_that("removeNA", {
+    expect_equal(removeNA(x = c(NA, 1, 2, NA, 3)), c(1, 2 ,3))
+})
+
+test_that("splitByComma", {
+    expect_equal(splitByComma(input.text = "Q1,Q2"), c("Q1", "Q2"))
+    expect_equal(splitByComma(input.text = ",Q1,Q2, ,,Q3,"), c("Q1", "Q2", "Q3"))
+    expect_equal(splitByComma(input.text = ",Q1(,2, 3,,), ,, Q2(3,4),,Q3,",
+                              ignore.commas.in.parentheses = TRUE),
+                 c("Q1(,2, 3,,)", "Q2(3,4)", "Q3"))
+})
+
+test_that("isIntegerValued", {
+    expect_true(isIntegerValued(x = c(-1, 0, 1, 2, 3, NA, Inf)))
```

Comment:
Infinity is an integer?  That cuts things down!  We might yet isolate its real value :-)

Reply:
Oops, I have changed the function to not recognise Inf as integer in 2fd7df8